sentry-delayed_job/lib/sentry/delayed_job/plugin.rb

```diff
@@ -91,12 +91,42 @@ def self.report?(job)
         job.attempts >= max_attempts
       end
 
+      def self.start_transaction(scope, env, contexts)
+        options = { name: scope.transaction_name, source: scope.transaction_source, op: OP_NAME }
+        transaction = Sentry.continue_trace(env, **options)
+        Sentry.start_transaction(transaction: transaction, custom_sampling_context: contexts, **options)
+      end
+
       def self.finish_transaction(transaction, status)
         return unless transaction
 
         transaction.set_http_status(status)
         transaction.finish
       end
+
+      def self.inject_trace_data(job)
+        # active job style is handled in the sentry-rails/active_job extension more generally
+        # if someone enqueues manually with some other job class, we cannot make assumptions unfortunately
+        payload_object = job.payload_object
+        return unless payload_object.is_a?(Delayed::PerformableMethod)
+
+        # we will add the trace data to args and remove it again
+        # this is hacky but it's the only reliable way to survive the YAML serialization/deserialization
+        payload_object.args << { sentry: Sentry.get_trace_propagation_headers }
+        job.payload_object = payload_object
+      end
+
+      def self.extract_trace_data(job)
+        payload_object = job.payload_object
+        return nil unless payload_object.is_a?(Delayed::PerformableMethod)
+
+        ind = payload_object.args.index { |a| a.is_a?(Hash) && a.key?(:sentry) }
+        return nil unless ind
+
+        env = payload_object.args[ind][:sentry]
+        payload_object.args.delete_at(ind)
+        env
```

Comment:
Can this work?

```suggestion
        target_payload = payload_object.args.find { |a| a.is_a?(Hash) && a.key?(:sentry) }
        return nil unless target_payload
        payload_object.args.delete(target_payload)
        target_payload[:sentry]
```

Reply:
that delete will do a find again under the hood and I'm not sure it works with an object like that.

Reply:
It works because `delete` uses `==` under the hood. And since the value here is a hash with string values, there should be no issue. I also have tested it with the test suites.